src/python_pachyderm/__init__.py

```diff
@@ -15,9 +15,16 @@
 from grpc import RpcError
 from .experimental import ExperimentalClient
 
+# Python version compatibility.
+try:
+    # >= 3.8
+    import importlib.metadata as metadata
```

Comment:
Does this read directly from `pyproject.toml`?

Reply:
Nope, packages keep their official version in an associated `METADATA` file that these modules attempt to read (this is how pip can report the version of installed packages without a lock file).